### ISSUE TITLE
[Question] Difference between stable-baselines3 and SpinUp

### ISSUE THREAD

**Issue by zhihanyang2022** [DLR-RM/stable-baselines3#452]
### Question

I have implemented DDPG, TD3 and SAC following SpinUp's tutorials. However, later I found that they do not perform as well as the stable-baselines3 implementations on some custom tasks.

So, my question is, what are some additional features used here that is NOT used in SpinUp (apart from auto entropy tuning for SAC and hyper-parameters)? I'm asking this because I noticed that the implementations here use SpinUp as a reference.

### Additional context

I have already parsed through the source code, but nothing special caught my eye. So the difference might be sometime very small. 

I have also checked my implementations by comparing to SpinUp's benchmark, so it's unlikely that my implementation has lots of bugs.

### Checklist

- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)

**Comment by araffin** [comment]
Hello,
which version of SB3 are you using and are you sure the hyperparameters are exactly the same?

**Comment by araffin** [comment]
>I have already parsed through the source code, but nothing special caught my eye. So the difference might be sometime very small.

have you checked your code again the spinning up one?
One small difference that has a great impact may be the proper handling of timeouts (max episode length).
The other difference is how often you do the gradient updates, for DDPG, TD3 and SAC, a gradient step is done after each call to `env.step()`.

**Comment by zhihanyang2022** [comment]
@araffin Thanks a lot for the quick response. It turns out that automatic entropy tuning was actually very important for my custom gym task. The reason was that the task involved sparse terminal reward, and using a fixed entropy coefficient (like 0.2 in spinup) caused the agent to be satisfied by just being random and without actual reaching the goal.
